tests/testthat/test-exercise.R

```diff
@@ -218,3 +218,27 @@ test_that("serialized exercises produce equivalent evaluate_exercise() results",
     env_vals(ex_eval_rmote$feedback$checker_args$envir_result)
   )
 })
+
+# filter_dependencies() ---------------------------------------------------
+
+test_that("filter_dependencies() excludes non-list knit_meta objects", {
+  ex <- mock_exercise(
+    user_code = deparse(quote(
+      htmltools::tagList(
+        htmltools::tags$head(htmltools::tags$style(".leaflet-container {backround:#FFF}")),
+        learnr::html_dependency_jquery()
+      )
+    ))
+  )
+
+  ex_res <- expect_silent(render_exercise(ex, new.env()))
+
+  ex_res_html_deps <- htmltools::htmlDependencies(ex_res$html_output)
+  # The head(style) dependency is dropped because it's not from a package
+  expect_equal(length(ex_res_html_deps), 1L)
+  # But we keep the dependency that came from a pkg
+  expect_equal(
+    ex_res_html_deps[[1]],
+    learnr::html_dependency_jquery()
```

Comment:
(Prolly from local copy/paste debugging)
```suggestion
    html_dependency_jquery()
```